src/dotnet/APIView/APIViewWeb/Languages/JavaLanguageService.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using ApiView;
+
+namespace APIViewWeb
+{
+    public class JavaLanguageService : ILanguageService
+    {
+        public string Name { get; } = "Java";
+
+        public string JarName = "java-api-listing-1.0.1.jar";
+
+        public bool IsSupportedExtension(string extension)
+        {
+            return string.Equals(extension, ".jar", comparisonType: StringComparison.OrdinalIgnoreCase);
```

Comment:
To be even more clear, we could enforce `-sources.jar`, so people don't upload compiled code. This isn't technically an extension though.